miden/tests/integration/stdlib/collections/mmr.rs

```diff
@@ -0,0 +1,168 @@
+use crate::{build_debug_test, build_test, helpers::crypto::init_merkle_leaves};
+use vm_core::{
+    crypto::merkle::{MerkleError, MerkleStore, MerkleTree, NodeIndex},
+    StarkField,
+};
+
+#[test]
+fn test_ilog2() {
+    let bit31 = "
+    use.std::collections::mmr
+
+    begin
+        push.2147483648
+        exec.mmr::ilog2_checked
+    end
+    ";
+
+    let test = build_test!(bit31, &[]);
+    test.expect_stack(&[31, 1 << 31]);
+
+    let bit31_and_one = "
+    use.std::collections::mmr
+
+    begin
+        push.2147483649
+        exec.mmr::ilog2_checked
+    end
+    ";
+
+    let test = build_test!(bit31_and_one, &[]);
+    test.expect_stack(&[31, 1 << 31]);
+
+    let bit16 = "
+    use.std::collections::mmr
+
+    begin
+        push.65536
+        exec.mmr::ilog2_checked
+    end
+    ";
+
+    let test = build_test!(bit16, &[]);
+    test.expect_stack(&[16, 1 << 16]);
+
+    let all_bits_from_16 = "
+    use.std::collections::mmr
+
+    begin
+        push.131071
+        exec.mmr::ilog2_checked
+    end
+    ";
+
+    let test = build_test!(all_bits_from_16, &[]);
+    test.expect_stack(&[16, 1 << 16]);
+
+    let one = "
+    use.std::collections::mmr
+
+    begin
+        push.1
+        exec.mmr::ilog2_checked
+    end
+    ";
+
+    let test = build_test!(one, &[]);
+    test.expect_stack(&[0, 1 << 0]);
+}
+
+#[test]
+fn test_mmr_get_single_peak() -> Result<(), MerkleError> {
+    // This test uses a single merkle tree as the only MMR peak
+    let leaves = &[1, 2, 3, 4];
+    let mut merkle_store = MerkleStore::new();
+    let merkle_root = merkle_store.add_merkle_tree(init_merkle_leaves(leaves))?;
+    let advice_stack: Vec<u64> = merkle_root.iter().map(StarkField::as_int).collect();
+
+    for pos in 0..(leaves.len() as u64) {
+        let source = format!(
+            "use.std::collections::mmr
+
+            begin
+                push.{num_leaves} push.1000 mem_store # leaves count
+                adv_push.4 push.1001 mem_storew dropw # MMR single peak
+
+                push.1000 push.{pos} exec.mmr::get
+            end",
+            num_leaves = leaves.len(),
+            pos = pos,
+        );
+
+        let test = build_debug_test!(source, &[], advice_stack, merkle_store.clone());
+        let leaf = merkle_store.get_node(merkle_root, NodeIndex::new(2, pos))?;
+
+        // the stack should be first the leaf element followed by the tree root
+        let stack: Vec<u64> = leaf
+            .iter()
+            .map(StarkField::as_int)
+            .rev()
+            .chain(merkle_root.iter().map(StarkField::as_int).rev())
+            .collect();
+        test.expect_stack(&stack);
+    }
+
+    Ok(())
+}
+
+#[test]
+fn test_mmr_get_two_peaks() -> Result<(), MerkleError> {
+    // This test uses two merkle trees for the MMR, one with 8 elements, and one with 2
+    let leaves1 = &[1, 2, 3, 4, 5, 6, 7, 8];
+    let leaves2 = &[9, 10];
```

Comment:
We also need a test case where leaves are something like this:
```Rust
let leaves1 = &[1, 2, 3, 4, 5, 6, 7, 8];
let leaves2 = &[9, 10];
let leaves3 = &[11];
```
So that we can test getting the singular leaf.

Reply:
added new test